### ISSUE TITLE
random: present but cannot be compiled

### ISSUE THREAD

**Issue by peternewman** [OpenLightingProject/ola#1477]
From config.log:

```
configure:6553: checking random usability
configure:6553: g++ -c -g -O2 -std=gnu++98  conftest.cpp >&5
In file included from /usr/include/c++/6/random:35:0,
                 from conftest.cpp:93:
/usr/include/c++/6/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support \
  ^~~~~
configure:6553: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "OLA"
| #define PACKAGE_TARNAME "ola"
| #define PACKAGE_VERSION "0.10.7"
| #define PACKAGE_STRING "OLA 0.10.7"
| #define PACKAGE_BUGREPORT "open-lighting@googlegroups.com"
| #define PACKAGE_URL ""
| #define PACKAGE "ola"
| #define VERSION "0.10.7"
| #define HAVE_DIRENT_H 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_NETINET_IN_H 1
| #define HAVE_ARPA_NAMESER_H 1
| #define HAVE_NETDB_H 1
| #define HAVE_RESOLV_H 1
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_ERRNO_H 1
| #define HAVE_ARPA_INET_H 1
| #define HAVE_BITS_SOCKADDR_H 1
| #define HAVE_FCNTL_H 1
| #define HAVE_FLOAT_H 1
| #define HAVE_LIMITS_H 1
| #define HAVE_MALLOC_H 1
| #define HAVE_NETINET_IN_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_SYS_FILE_H 1
| #define HAVE_SYS_IOCTL_H 1
| #define HAVE_SYS_SOCKET_H 1
| #define HAVE_SYS_TIME_H 1
| #define HAVE_SYS_TIMEB_H 1
| #define HAVE_SYSLOG_H 1
| #define HAVE_TERMIOS_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_ASM_TERMIOS_H 1
| #define HAVE_ASSERT_H 1
| #define HAVE_DLFCN_H 1
| #define HAVE_ENDIAN_H 1
| #define HAVE_EXECINFO_H 1
| #define HAVE_LINUX_IF_PACKET_H 1
| #define HAVE_MATH_H 1
| #define HAVE_NET_ETHERNET_H 1
| #define HAVE_STROPTS_H 1
| #define HAVE_SYS_PARAM_H 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_UIO_H 1
| #define HAVE_SYSEXITS_H 1
| /* end confdefs.h.  */
| #include <stdio.h>
| #ifdef HAVE_SYS_TYPES_H
| # include <sys/types.h>
| #endif
| #ifdef HAVE_SYS_STAT_H
| # include <sys/stat.h>
| #endif
| #ifdef STDC_HEADERS
| # include <stdlib.h>
| # include <stddef.h>
| #else
| # ifdef HAVE_STDLIB_H
| #  include <stdlib.h>
| # endif
| #endif
| #ifdef HAVE_STRING_H
| # if !defined STDC_HEADERS && defined HAVE_MEMORY_H
| #  include <memory.h>
| # endif
| # include <string.h>
| #endif
| #ifdef HAVE_STRINGS_H
| # include <strings.h>
| #endif
| #ifdef HAVE_INTTYPES_H
| # include <inttypes.h>
| #endif
| #ifdef HAVE_STDINT_H
| # include <stdint.h>
| #endif
| #ifdef HAVE_UNISTD_H
| # include <unistd.h>
| #endif
| #include <random>
configure:6553: result: no
configure:6553: checking random presence
configure:6553: g++ -E  conftest.cpp
configure:6553: $? = 0
configure:6553: result: yes
configure:6553: WARNING: random: present but cannot be compiled
configure:6553: WARNING: random:     check for missing prerequisite headers?
configure:6553: WARNING: random: see the Autoconf documentation
configure:6553: WARNING: random:     section "Present But Cannot Be Compiled"
configure:6553: WARNING: random: proceeding with the compiler's result
configure:6553: checking for random
configure:6553: result: no
```

https://groups.google.com/forum/#!topic/open-lighting/2ICj2yAMhfE

We could extend the configure checks we've already got around cppunit/protobuf to also force us to C++11 if random.h requires it too. We currently fall back to a different random function that's not as good. However we're not (currently) using it for any security stuff, so I don't believe it's too critical.

**Comment by peternewman** [comment]
To test:
`watch -n 0.1 "./examples/ola_rdm_get -u 0 --uid 7a70:ffffff05 sensor_value 0 | grep Present | sed -e 's/Present Value://' | tee -a sensorvalues.log"`

Then:
`sort -n sensorvalues.log | uniq -c | less`

**Comment by peternewman** [comment]
Closed via #1498 .
